eventuate-core/src/test/scala/com/rbmhtechnology/eventuate/EventsourcedActorTestKit.scala

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright 2015 - 2016 Red Bull Media House GmbH <http://www.redbullmediahouse.com> - all rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rbmhtechnology.eventuate
+
+import akka.actor.ActorRef
+import akka.actor.ActorSystem
+import akka.actor.Props
+import akka.testkit.TestActorRef
+import akka.testkit.TestKit
+import akka.testkit.TestProbe
+import akka.util.Timeout
+import com.rbmhtechnology.eventuate.DurableEvent
+import com.rbmhtechnology.eventuate.EventSourcedActorTestKit.Persist
+import com.rbmhtechnology.eventuate.EventsourcedActor
+import com.rbmhtechnology.eventuate.EventsourcingProtocol.WriteSuccess
+import com.rbmhtechnology.eventuate.EventsourcingProtocol._
+import com.typesafe.config.ConfigFactory
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.FeatureSpecLike
+import org.scalatest.Matchers
+import org.scalatest.words.ShouldVerb
+
+import scala.concurrent.Await
+import scala.concurrent.Awaitable
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.duration._
+import scala.reflect.ClassTag
+import scala.util.Failure
+import scala.util.Success
+import scala.util.Try
+
+object EventSourcedActorTestKit {
+
+  /** actor system all actors under test run in */
+  val actorSystem = ActorSystem("test", ConfigFactory.parseString("eventuate.log.write-timeout = 1s"))
+
+  /** Simulate persisting 'event'. Actors this is sent to must implement [[ExplicitOnEventCall]] */
+  case class Persist(event: Any)
+
+}
+
+/**
+  * Actor tests need to implement this trait as well as [[akka.testkit.TestKit]]. A default [[ActorSystem]] is given
+  * by companion object.  Example for usage of this trait: <br>
+  *
+  * {{{ class MyActorTest extends TestKit(ActorTest.actorSystem)  with ActorTest with GivenWhenThen { ... } }}}
+  *
+  * Usage: <br>
+  * - instantiate actors to test via initEventSourcedActorAsTestActorRef or initEventSourcedActor. Former returns an
+  * [[TestActorRef]] and thus allows to test _.underlyingActor. <br>
+  * - either tell returned [[ActorRef]]/[[TestActorRef]] a command or explicitly call onEvent via #callOnEvent <br>
+  * - query 'peristed' events via #expectEvent or #expectEvents <br><br>
+  *
+  * All actor's under test must have #eventLog as their eventLog. Provided event log can be viewed as a stash. Events
```

Comment:
The provided event log can be viewed as a **queue**. The first written event can be expected first (FIFO). Consequently, do not use operations named *stash* and *pop* to refer to operations on the test event log (used at several places in code and API docs).